lib/io/QuadSinkHdt.ts

```diff
@@ -1,85 +1,123 @@
 import * as Docker from 'dockerode';
-import type { IQuadSinkFileOptions } from './QuadSinkFile';
-import { QuadSinkFile } from './QuadSinkFile';
-import { transformToHdt, pullHdtCppDockerImage } from './rfdhdtDockerUtil';
+import { QuadSinkFile, type IQuadSinkFileOptions } from './QuadSinkFile';
+
+const HDTCPP_MOUNT_PATH = '/tmp/convert';
+const HDTCPP_DOCKER_IMAGE = 'rdfhdt/hdt-cpp:latest';
+const HDTCPP_FORMATS = new Map<string, string>([
+  [ 'application/n-quads', 'nquad' ],
+  [ 'application/n-triples', 'ntriples' ],
+  [ 'text/turtle', 'turtle' ],
+  [ 'application/rdf+xml', 'rdfxml' ],
+  [ 'text/n3', 'n3' ],
+]);
 
 export class QuadSinkHdt extends QuadSinkFile {
-  private readonly files: Set<string> = new Set();
-  private readonly deleteSourceFiles: boolean;
-  private readonly errorFileDockerRfdhdt: WriteStream;
-  private readonly poolSize: number;
+  private readonly generateIndexes: boolean;
+  private readonly conversionProgress: boolean;
+  private readonly removeSourceFiles: boolean;
+  private readonly conversionConcurrency: number;
+
+  private readonly docker: Docker;
+  private readonly filesToConvert: Set<string>;
 
-  public constructor(
-    options: IQuadSinkFileOptions,
-    poolSize = 1,
-    deleteSourceFiles = false,
-    errorFileDockerRfdhdt = './error_log_docker_rfdhdt.txt',
-  ) {
+  public constructor(options: IQuadSinkHdtOptions) {
     super(options);
-    this.deleteSourceFiles = deleteSourceFiles;
-    this.errorFileDockerRfdhdt = createWriteStream(errorFileDockerRfdhdt);
-    this.poolSize = poolSize;
+    if (!HDTCPP_FORMATS.has(options.outputFormat)) {
+      throw new Error(`Unsupported HDT output format ${options.outputFormat}`);
+    }
+    this.generateIndexes = options.generateIndexes ?? false;
+    this.conversionProgress = options.conversionProgress ?? false;
+    this.removeSourceFiles = options.removeSourceFiles ?? true;
+    this.conversionConcurrency = options.conversionConcurrency ?? 1;
+    this.docker = new Docker();
+    this.filesToConvert = new Set();
   }
 
-  public async push(iri: string, quad: RDF.Quad): Promise<void> {
-    const path = Path.join('./', this.getFilePath(iri));
-    await super.push(iri, quad);
+  protected getFilePath(iri: string): string {
+    const path = super.getFilePath(iri);
+    this.filesToConvert.add(resolve(path));
+    return path;
+  }
+
+  protected async pullDockerImage(): Promise<void> {
+    await new Promise<void>((resolve, reject) => this.docker.pull(HDTCPP_DOCKER_IMAGE, {}, (error, result) => {
+      if (error) {
+        reject(error);
+      }
+      this.docker.modem.followProgress(result!, error => error ? reject(error) : resolve());
+    }));
+  }
 
-    // Add files with the defined extension to the list to be transformed
-    if (this.fileExtension !== undefined && path.includes(this.fileExtension)) {
-      this.files.add(path);
+  protected async convertSingleFile(rdfFilePath: string): Promise<void> {
```

Comment:
I like to have in the comment the actual equivalent of the docker commands, since we simply puppet docker.